docs/Lexicon.md

```diff
@@ -0,0 +1,562 @@
+transitively consider dependency edges in the file that defines the used
+name when incremental compilation is enabled. A cascading dependency is much
+safer to produce than its private counterpart, but it comes at the cost of
+increased usage of compilation resources - even if those resources are being
+wasted on rebuilding a file that didn't actually require rebuilding.
+See [DependencyAnalysis.rst](DependencyAnalysis.rst).
+
+## Clang importer
+
+The part of the compiler that reads C and Objective-C declarations and
+exposes them as Swift. Essentially contains a small instance of Clang
+running inside the Swift compiler, which is also used during IRGen.
+
+## conformance
+
+A construct detailing how a particular type conforms to a particular
+protocol. Represented in the compiler by the ProtocolConformance type at
+the AST level. See also [witness table](#witness-table).
+
+## contextual type
+
+1. The expected type for a Swift sub-expression based on the rest of the
+   statement. For example, in the statement `print(6 * 9)`, the contextual
+   type of the expression `6 * 9` is `Any`.
+2. The type of a value or declaration from inside a potentially generic
+   context. This type may contain [archetypes](#archetype) and cannot be
+   used directly from outside the context. Compare with [interface type](#interface-type).
+
+## customization point
+
+Informal term for a protocol requirement that has a default implementation,
+i.e. one that conforming types don't *have* to implement but have the option
+to "customize".
+
+## dependency sink
+
+Any request that uses a matching dependency source to write dependency
+edges into the referenced name trackers. For example, a request that
+performs direct lookup will write the name being looked up into the
+name tracker associated with the file that issued the lookup request.
+The request evaluator automatically determines the appropriate tracker
+for a dependency sink to write into based on the current active
+[dependency source](#dependency-source) request.
+
+## dependency source
+
+Any request that defines a scope under which reference dependencies may be
+registered. For example, a request to type check an entire file is a
+dependency source. Dependency sources are automatically managed by the
+request evaluator as request evaluation proceeds. Dependency sources provide
+one half of the necessary information to complete a full dependency edge.
+The other half is provided by corresponding
+[dependency sink](#dependency-sink) requests.
+
+## DI (definite initialization / definitive initialization)
+
+The feature that no uninitialized variables, constants, or properties will
+be read by a program, or the analysis pass that operates on SIL to
+guarantee this. This was 
+[discussed on Apple's Swift blog](https://developer.apple.com/swift/blog/?id=28).
+
+## DNM
+
+"Do not merge". Placed in PR titles where discussion or analysis is still
+ongoing.
```

Comment:
Outdated? [*Draft pull requests*](https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/)